system/lib/emmalloc.c

```diff
@@ -97,6 +97,13 @@ typedef struct Region
   uint32_t _at_the_end_of_this_struct_size; // do not dereference, this is present for convenient struct sizeof() computation only
 } Region;
 
+typedef struct RootRegion
+{
+  uint32_t size;
+  struct RootRegion *next;
+  uint8_t* endPtr;
+} RootRegion;
```

Comment:
Maybe this struct could also like to have a comment that explains its purpose, something like

```
// Each memory region block starts with a RootRegion at the beginning of the memory region. The RootRegion specifies the size of the region block, and forms a linked list of all RootRegions in the program.
```